ztp/policygenerator/policyGen/policyBuilder.go

```diff
@@ -86,6 +87,25 @@ func (pbuilder *PolicyBuilder) Build(policyGenTemp utils.PolicyGenTemplate) (map
 					if err != nil {
 						return policies, err
 					}
+					//set to default remediationAction
+					remediationActionVal := acmPolicy.Spec.RemediationAction
+					if policyGenTemp.Spec.RemediationAction != "" {
+						// PGT level remediationAction overrides default
+						remediationActionVal = policyGenTemp.Spec.RemediationAction
+						if sFile.RemediationAction != "" && sFile.RemediationAction != policyGenTemp.Spec.RemediationAction {
+							// sourceFile remediationAction overrides the PGT level remediationAction
+							remediationActionVal = sFile.RemediationAction
+							remediationActionMap[sFile.PolicyName] = sFile.RemediationAction
+						}
+					} else if sFile.RemediationAction != "" {
```

Comment:
I guess sFile.RemediationAction need to be checked first as it has the higher priority to consider.